lib/src/wolt_modal_sheet.dart

```diff
@@ -352,18 +352,15 @@ class WoltModalSheetState extends State<WoltModalSheet> {
   @override
   void initState() {
     super.initState();
-    widget.pageListBuilderNotifier.addListener(() {
-      // Update the page list whenever the notifier changes. This is useful when the page list is
-      // set in a declarative way using the WoltModalSheetRoute.
-      setState(() {
-        final pages = widget.pageListBuilderNotifier.value(context);
-        assert(
-            pages.isNotEmpty, 'pageListBuilder must return a non-empty list.');
-        _pages
-          ..clear()
-          ..addAll(pages);
-      });
-    });
+    widget.pageListBuilderNotifier
+        .addListener(_onPageListBuilderNotifierValueUpdated);
+  }
+
+  @override
+  void dispose() {
+    widget.pageListBuilderNotifier
+        .removeListener(_onPageListBuilderNotifierValueUpdated);
```

Comment:
Errors fixes are excellent for writing test.  Also please execute `dart format .`

Reply:
Added tests: 13afb0ed5117ed7e23d6e9e9e90a2d6e82e27c5f 
Formatted: 0f8596d45adfbedba459dae1939b79e238069f42

Reply:
Awesome!